src/desktop/screencast.rs

```diff
@@ -417,3 +423,11 @@ impl<'a> std::ops::Deref for Screencast<'a> {
         &self.0
     }
 }
+
+impl SessionPortal for Screencast<'_> {}
+
+/// Defines which portals session can be used in a screen-cast.
+pub trait ScreencastSession: Serialize + zvariant::Type + Debug {}
+
+impl ScreencastSession for Session<'_, Screencast<'_>> {}
+impl ScreencastSession for Session<'_, RemoteDesktop<'_>> {}
```

Comment:
```suggestion
pub trait HasScreencastSession {}

impl HasScreencastSession for Screencast<'_> {}
impl HasScreencastSession for RemoteDesktop<'_> {}
```

and then use `Session<'_, impl HasScreencastSession>` for the parameters. That way it's guaranteed to be a session and you don't need all the trait bounds

Reply:
```diff
diff --git a/src/desktop/screencast.rs b/src/desktop/screencast.rs
index e427aa1..ae2eda3 100644
--- a/src/desktop/screencast.rs
+++ b/src/desktop/screencast.rs
@@ -45,7 +45,7 @@ use std::{collections::HashMap, fmt::Debug, os::fd::OwnedFd};
 
 use enumflags2::{bitflags, BitFlags};
 use futures_util::TryFutureExt;
-use serde::{Deserialize, Serialize};
+use serde::Deserialize;
 use serde_repr::{Deserialize_repr, Serialize_repr};
 use zbus::zvariant::{self, DeserializeDict, SerializeDict, Type, Value};
 
@@ -307,7 +307,7 @@ impl<'a> Screencast<'a> {
     #[doc(alias = "OpenPipeWireRemote")]
     pub async fn open_pipe_wire_remote(
         &self,
-        session: &impl ScreencastSession,
+        session: &Session<'_, impl HasScreencastSession>,
     ) -> Result<OwnedFd, Error> {
         // `options` parameter doesn't seems to be used yet
         // see https://github.com/flatpak/xdg-desktop-portal/blob/master/src/screen-cast.c#L812
@@ -341,7 +341,7 @@ impl<'a> Screencast<'a> {
     #[doc(alias = "SelectSources")]
     pub async fn select_sources(
         &self,
-        session: &impl ScreencastSession,
+        session: &Session<'_, impl HasScreencastSession>,
         cursor_mode: CursorMode,
         types: BitFlags<SourceType>,
         multiple: bool,
@@ -382,7 +382,7 @@ impl<'a> Screencast<'a> {
     #[doc(alias = "Start")]
     pub async fn start(
         &self,
-        session: &impl ScreencastSession,
+        session: &Session<'_, impl HasScreencastSession>,
         identifier: &WindowIdentifier,
     ) -> Result<Request<Streams>, Error> {
         let options = StartCastOptions::default();
@@ -427,7 +427,7 @@ impl<'a> std::ops::Deref for Screencast<'a> {
 impl SessionPortal for Screencast<'_> {}
 
 /// Defines which portals session can be used in a screen-cast.
-pub trait ScreencastSession: Serialize + zvariant::Type + Debug {}
+pub trait HasScreencastSession: SessionPortal {}
 
-impl ScreencastSession for Session<'_, Screencast<'_>> {}
-impl ScreencastSession for Session<'_, RemoteDesktop<'_>> {}
+impl HasScreencastSession for Screencast<'_> {}
+impl HasScreencastSession for RemoteDesktop<'_> {}
```